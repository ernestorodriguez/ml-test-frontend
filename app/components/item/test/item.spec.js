const React = require('react');
const Item = require('../');
const { shallow } = require('enzyme');
const reactRouter = require('react-router-dom');

jest.mock('react-router');

describe('Item component', () => {
    afterAll(() => {
        reactRouter.mockRestore();
    });

    test('must show header with and input and button for search and send query if button is press', () => {
        const preventDefault = jest.fn();
        const push = jest.fn();

        reactRouter.useHistory.mockImplementation(() => ({ push }));

        const wrapper = shallow(<Item id={'MLA123'} />);

        wrapper.find('[data-js="item"]').simulate('click', {
            preventDefault
        });
        expect(wrapper).toMatchSnapshot();
        expect(preventDefault.mock.calls.length).toBe(1);
        expect(reactRouter.useHistory.mock.calls.length).toBe(1);
        expect(push).toBeCalledWith('/items/MLA123');
    });
});
