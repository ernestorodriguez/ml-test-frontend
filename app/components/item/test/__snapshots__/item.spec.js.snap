// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component must show header with and input and button for search and send query if button is press 1`] = `
<section
  className="item"
>
  <form
    action="/items/MLA123"
  >
    <button
      data-js="item"
      onClick={[Function]}
      type="submit"
    >
      <h2>
        Item
      </h2>
    </button>
  </form>
</section>
`;
